src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -340,4 +348,49 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema)(im
       }
     }
   }
+
+  def importPFB(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]],
+                workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest): Future[PerRequestMessage] = {
+    def callArrow = {
+      logger.debug("calling arrow")
+      pipeline {
+        Post(avroToRawlsURL, requestContext.request.entity)
+      }
+    }
+
+    def callRawls(arrowResponse: HttpResponse) = {
+      logger.debug("calling rawls")
+      pipeline {
+        Post(FireCloudDirectiveUtils.encodeUri(Rawls.entityPathFromWorkspace(workspaceNamespace, workspaceName) + "/batchUpsert"), arrowResponse.entity)
+      }
+    }
+
+    val pfbUrl = new URL(pfbRequest.url.replace(" ", "%20"))
```

Comment:
hm, is there a less manual solution for this, something with `URLEncoder.encode` or `FireCloudDirectiveUtils.encodeUri`, or one of the `URI` methods? Alternately, assuming the end user properly encoded a uri, how did it get decoded by the time it reaches here? I'm worried the manual handling of spaces here will miss some other tricky character.

Reply:
Agree. I copied this line from `importBagit` above this in the file (there are some strong similarities between these two features). I'll test out using `FireCloudDirectiveUtils.encodeUri` instead.

Reply:
Also, I'm not even using this value in the request to Arrow. Oops! Need to fix that.

Reply:
Unless this should be Arrow's responsibility. 🤔

Reply:
Turns out `FireCloudDirectiveUtils.encodeUri` does _not_ handle this. I was able to easily do this within Arrow which I think should be doing this anyway so no need for orch to do it.

The URL is a string in the JSON payload, not on the query string, so it can have any value. Historically, we encountered this in production with the bagit import so I figure it's a good idea to do it for PFB import as well.

Reply:
sounds good - as long as whatever we report back to the user is helpful, I'm totally find if it happens within Arrow.